### PR TITLE
fix graphite mysql password issue

### PR DESCRIPTION
**Pull request by DrRayWang**
fix the issue coursed by graphite mysql password access. It should be accessed directly from chef vaulted items, not from default data bag 'configs'.

### CODE REVIEW

File: cookbooks/bcpc/recipes/graphite.rb [hunk 1]
```diff
@@ -175,8 +175,8 @@
     block do
         if not system "mysql -uroot -p#{get_config('mysql-root-password')} -e 'SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = \"#{node['bcpc']['graphite_dbname']}\"'|grep \"#{node['bcpc']['graphite_dbname']}\"" then
             %x[ mysql -uroot -p#{get_config('mysql-root-password')} -e "CREATE DATABASE #{node['bcpc']['graphite_dbname']};"
-                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'%' IDENTIFIED BY '#{get_config('mysql-graphite-password')}';"
-                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'localhost' IDENTIFIED BY '#{get_config('mysql-graphite-password')}';"
+                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'%' IDENTIFIED BY '#{get_config('password','mysql-graphite','os')}';"
```
Comment (cbaenziger) [inline cookbooks/bcpc/recipes/graphite.rb]: I think this will silently fail as we are using `get_config`; this should likely be [`get_config!`](https://github.com/bloomberg/chef-bach/blob/9414031641074410f52362aea3ede6e38ccad17d/cookbooks/bcpc/libraries/utils.rb#L118-L122) to `raise` if the

File: cookbooks/bcpc/recipes/graphite.rb [hunk 2]
```diff
@@ -175,8 +175,8 @@
...
             %x[ mysql -uroot -p#{get_config('mysql-root-password')} -e "CREATE DATABASE #{node['bcpc']['graphite_dbname']};"
-                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'%' IDENTIFIED BY '#{get_config('mysql-graphite-password')}';"
-                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'localhost' IDENTIFIED BY '#{get_config('mysql-graphite-password')}';"
+                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'%' IDENTIFIED BY '#{get_config!('password','mysql-graphite','os')}';"
+                mysql -uroot -p#{get_config('mysql-root-password')} -e "GRANT ALL ON #{node['bcpc']['graphite_dbname']}.* TO '#{get_config('mysql-graphite-user')}'@'localhost' IDENTIFIED BY '#{get_config!('password','mysql-graphite','os')}';"
```
Comment (amithkanand) [inline cookbooks/bcpc/recipes/graphite.rb]: @DrRayWang: Did you test it after changing the method call to `get_config!`? 

This will not work as `get_config!` doesn't have correct method signatures.

File: cookbooks/bcpc/recipes/graphite.rb (hunk 1 above)
Comment (DrRayWang) [inline cookbooks/bcpc/recipes/graphite.rb]: This is an issue covered by [issue 294](https://github.com/bloomberg/chef-bach/issues/294), will be fixed in a separate PR.

**Comment by amithkanand** [comment]
LGTM
